pallene/checker.lua

```diff
@@ -391,14 +463,52 @@ function FunChecker:check_function(lambda, func_typ)
     end)
 end
 
+function FunChecker:check_and_declare_var(decl, exp)
+    local typ
+    typ, exp = self:check_initializer_exp(decl,
+        exp, "declaration of local variable %s",
+        decl.name)
+    self:add_local(decl.name, typ)
+    decl._name = self.p.symbol_table:find_symbol(decl.name)
+    return exp
```

Comment:
This code isn't implementing the scope correctly. The variables being declared should NOT be in scope for subsequent initializers.

Please consider the following test case. It should be allowed but currently it is gives a type error:

```
function test()
    local a: integer = 10
    local a: float, b: integer = 3.14, a
end
```

Reply:
I didn't think about this case. Already fixed it.